node/node.go

```diff
@@ -258,6 +259,10 @@ func New(host p2p.Host, consensusObj *consensus.Consensus, db ethdb.Database) *N
 		node.AddFaucetContractToPendingTransactions()
 		node.CurrentStakes = make(map[common.Address]*structs.StakeInfo)
 		node.AddStakingContractToPendingTransactions() //This will save the latest information about staked nodes in current staked
+
+		// TODO(minhdoan): Think of a better approach to deploy smart contract.
+		// This is temporary for demo purpose.
+		node.AddLotteryContract()
```

Comment:
This is indeed a problem. (Not to be fixed in this PR) but thanks for calling it out.  My earlier investigation had brought me to this: https://gist.github.com/skabbass1/f37e35998723cd2bb8e21a01fefbb0d7#file-deploy-go But I am not sure whether it can work with what we are building.

Reply:
Offline discussed